### PR TITLE
Bring ros2 branch closer to melodic-devel branch and ROS 2 conventions

### PR DESCRIPTION
**Pull request by mikaelarguedas**
This brings `class_loader` closer to ROS2 styls as well as the current ROS1 melodic version. There is still a significant difference between the ROS1 and ROS2 version but this PR should allow to more easily port patches from one to the next and allows to run ROS2 linters and use stricter compiler options:

Main things in this PR: 
- Respect ROS 2 style (pulled from melodic branch)
- Update ROS 2-only tests to the ROS 2 style as well
- Enable linters and compilation with -Wpedantic (copyright test disabled for now as the linter doesnt support copyright in `/**` comment block)
- Rename all the headers from .h to .hpp and update the include statements accordingly

Reviewing on a per commit basis would be preferred as well as the code shuffled around will not appear in the per commit diff.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3956/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1075/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3291/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4058/)

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
@dirk-thomas @wjwwood I think this is ready for review.

New round of CI for this and connected PRs:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3961/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1077/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3295/) (unrelated osrf_pycommon test failures)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4062/) (unrelated osrf_pycommon test failures)

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,6 +44,13 @@ if(WIN32)
 endif()
 
 if(BUILD_TESTING)
+  # TODO(mikaelarguedas) add copyright test once the linter accepts /** block comments
```
Comment (dirk-thomas) [inline CMakeLists.txt]: This comment looks like a duplicate. Maybe you wanted to mentioned something different here.

File: include/class_loader/class_loader.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef CLASS_LOADER__CLASS_LOADER_H_
+#define CLASS_LOADER__CLASS_LOADER_H_
+
+// *INDENT-OFF* (prevent uncrustify from adding indention below)
+#warning Including header <class_loader/class_loader.h> is deprecated, \
+include <class_loader/class_loader.hpp> instead.
```
Comment (dirk-thomas) [inline include/class_loader/class_loader.h]: Is there a way to indent the wrapped line and still keep the linters happy?

File: include/class_loader/class_loader.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef CLASS_LOADER__CLASS_LOADER_H_
+#define CLASS_LOADER__CLASS_LOADER_H_
+
+// *INDENT-OFF* (prevent uncrustify from adding indention below)
```
Comment (dirk-thomas) [inline include/class_loader/class_loader.h]: The `INDENT-OFF` is not reverted anywhere below.

**Comment by dirk-thomas** [comment]
Overall looks fine to me (haven't looked too closely). Also some of my comments might be invalid since changes from earlier commits have been changed / reverted in later commits.

File: include/class_loader/class_loader.h (hunk 2 above)
Comment (wjwwood) [reply include/class_loader/class_loader.h]: When I tried to do that the linter wants to put a tonne of whitespace in, like >8 spaces and that shows in the warning text unfortunately.

File: CMakeLists.txt (hunk 1 above)
Comment (mikaelarguedas) [reply CMakeLists.txt]: Sorry I noticed this after the fact and removed it in a later commit: https://github.com/ros/class_loader/pull/80/commits/877bcc263696b36d641543a13b9ab5ba8b9e83d2. Thanks for pointing it out though!

**Comment by mikaelarguedas** [comment]
Thanks @dirk-thomas for taking the time to review! I know it's a big diff pretty painful to review

**Pull request merged**